.github/workflows/test.yml

```diff
@@ -0,0 +1,36 @@
+name: CI
+
+on:
+  workflow_call:
+    inputs:
+      npm_command:
+        description: 'NPM command to run'
+        type: string
+        required: true
+        default: 'install'
+jobs:
+  test:
+    strategy:
+      fail-fast: false # prevent test to stop if one fails
+      matrix:
+        node-version: [18.x, 20.x]
+        os: [ubuntu-latest, windows-latest, macos-latest]
```

Comment:
what if we also expanded the matrix based on the individual tests? That way, each test would run in it's own runner asynchronously, and would also make it easier to see which test failed.

Reply:
That will not be beneficial as there will be too runners spawn and github has a limit on runners, you will end up having lot runners waiting

Reply:
Good to know! What is the limit? We've not run into that limit [here](https://github.com/TugboatQA/images/actions/runs/11300906163). Say the limit is 50, even if we hit the limit and had to wait, wouldn't that be faster?

Reply:
I see the limit in macos runner images, I think they give you at most 2/3 macos runner for time